apps/functions-subscription/src/main.ts

```diff
@@ -1,10 +1,50 @@
+import { pipe } from 'fp-ts/lib/function';
+import * as TE from 'fp-ts/lib/TaskEither';
 import { app } from '@azure/functions';
+import { CosmosClient } from '@azure/cosmos';
 import { makeInfoHandler } from './adapters/azure/functions/info';
+import { parseConfig } from './config';
+import { makeSubscriptionCosmosContainer } from './adapters/azure/cosmosdb/subscription';
+import { makeSubscriptionRequestEventHubProducer } from './adapters/azure/eventhubs/subscription-request';
+import { EventHubProducerClient } from '@azure/event-hubs';
 
+// the application entry-point
 // eslint-disable-next-line functional/no-expression-statements
-app.http('info', {
-  methods: ['GET'],
-  authLevel: 'anonymous',
-  handler: makeInfoHandler({}),
-  route: 'info',
-});
+pipe(
+  TE.Do,
+  TE.apS('config', TE.fromEither(parseConfig(process.env))),
+  TE.bind('cosmosDB', ({ config }) =>
+    TE.of(new CosmosClient(config.subscription.cosmosdb.connectionString)),
```

Comment:
nit: since these are global singletons and we're in the integration layer, we don't need to pass theme into a pipe, it suffices to assign a value at top level. as they should be refactored to throw in this case, just assure that the exception is handled correctly. this cuts some code.

Reply:
Ok, I'll try to see if doing that reduces some code 👍

Reply:
Done ✅